usr/share/rear/lib/global-functions.sh

```diff
@@ -558,65 +619,135 @@ mount_url() {
-            rm -rf /var/cache/davfs2/*outputfs*
-
-	    ;;
-        (var)
-            local var=$(url_host $url)
-            umount_cmd="${!var} $mountpoint"
+        (*)
+            # Schemes that actually need nontrivial umount are handled below.
+            # We do not handle them in the default branch because in the case of iso:
+            # it depends on the current workflow whether umount is needed or not.
+            :
+    esac
 
-            Log "Unmounting with '$umount_cmd'"
-            $umount_cmd
-            StopIfError "Unmounting failed."
+    # umount_url() is a wrapper that takes care of exit tasks and error handling and mountpoint cleanup.
+    # Therefore it also determines if exit task and mountpoint handling is required and returns early if not.
+    # The actual umount job is performed inside perform_umount_url().
+    # We do not request lazy umount here because we want umount errors to be reliably reported.
+    perform_umount_url $url $mountpoint || Error "Unmounting '$mountpoint' failed."
 
-            RemoveExitTask "umount -f $v '$mountpoint' >&2"
-            return 0
+    RemoveExitTask "perform_umount_url '$url' '$mountpoint' y"
+
+    remove_temporary_mountpoint '$mountpoint' && RemoveExitTask "remove_temporary_mountpoint '$mountpoint'"
+    return 0
+}
+
+### Unmount url $1 at mountpoint $2 [ lazily if $3 is set and normal unmount fails ]
+function perform_umount_url() {
+    local url=$1
+    local mountpoint=$2
+    local lazy=${3:-}
+
+    case $(url_scheme $url) in
+        (sshfs)
+            # does ftpfs need this special case as well?
+            fusermount -u ${lazy:+'-z'} $mountpoint
+            ;;
+        (davfs)
+            umount_davfs $mountpoint $lazy
+            ;;
+        (var)
+            local var
+            var=$(url_host $url)
+            Log "Unmounting with '${!var} $mountpoint'"
+            # lazy unmount not supported with custom umount command
+            ${!var} $mountpoint
             ;;
+        (*)
+            # usual umount command
+            umount_mountpoint $mountpoint $lazy
     esac
+    # The switch above must be the last statement in this function and the umount commands must be
+    # the last commands (or part of) in each branch. This ensures proper exit code propagation
+    # to the caller even when set -e is used.
+}
 
-    umount_mountpoint $mountpoint
-    StopIfError "Unmounting '$mountpoint' failed."
-
-    RemoveExitTask "umount -f $v '$mountpoint' >&2"
-    return 0
+### Helper which unmounts davfs mountpoint $1 and cleans up the cache,
+### performing lazy unmount if $2 is nonempty and normal unmount fails.
+function umount_davfs() {
+    local mountpoint=$1
+    local lazy="${2:-}"
+    if umount_mountpoint $mountpoint ; then
+        # Wait for 3 sek. then remove the cache-dir /var/cache/davfs
+        sleep 30
+        # ToDo: put in here the cache-dir from /etc/davfs2/davfs.conf
```

Comment:
Could you by the way change that `ToDo` into `TODO`
because at least my `vi` with its default settings
shows `TODO` (and `FIXME`) highlighted yellow
but no hightlighting for `ToDo` (or `FixMe`).

Reply:
will do

Reply:
Done via
https://github.com/rear/rear/pull/2625/commits/22302c52d336db11937f384d85b838c165050308
@pcahyna thank you!